CRM/Contact/BAO/GroupContactCache.php

```diff
@@ -67,17 +67,16 @@ public static function check($groupIDs) {
    */
   public static function groupRefreshedClause($groupIDClause = NULL, $includeHiddenGroups = FALSE) {
     $smartGroupCacheTimeout = self::smartGroupCacheTimeout();
-    $now = CRM_Utils_Date::getUTCTime();
+    $expiryTime = date_format(date_create(CRM_Utils_Date::getUTCTime(-($smartGroupCacheTimeout * 60))), 'Y-m-d H:i:s');
 
     $query = "
 SELECT  g.id
 FROM    civicrm_group g
 WHERE   ( g.saved_search_id IS NOT NULL OR g.children IS NOT NULL )
 AND     g.is_active = 1
 AND     ( g.cache_date IS NULL OR
-          ( TIMESTAMPDIFF(MINUTE, g.cache_date, $now) >= $smartGroupCacheTimeout ) OR
-          ( $now >= g.refresh_date )
-        )
+          g.cache_date <= '$expiryTime'
```

Comment:
OK so this does 2 things
1) it replaces the non-indexed way of referring to the cache_date field with an indexed one.
2) if removes the reference to refresh_date

I haven't decided what I think about removing refresh date yet but the former change needs to happen